libats-slick/src/test/scala/com/advancedtelematic/libats/slick/db/SlickExtensionsSpec.scala

```diff
@@ -42,4 +43,29 @@ class SlickExtensionsSpec extends FunSuite with Matchers with ScalaFutures with
 
     f.futureValue shouldBe Book(10, "Some book")
   }
+
+  test("resultHead on a Query returns the error in arg") {
+    val f = db.run(books.filter(_.id === 20l).resultHead(Error))
+    f.failed.futureValue shouldBe Error
+  }
+
+  test("maybeFilter uses filter if condition is defined") {
+    val f = for {
+      _ <- db.run(books += Book(30, "Some book", Option("30 some code")))
+      result <- db.run(books.maybeFilter(_.id === Option(30l)).result)
```

Comment:
Well it wasn't so much a question, as me being surprised of this behaviour of `===`, and it took me a while to even figure out the types. But this is fine, just a bit confusing at first :)